src/window_gui.h

```diff
@@ -260,8 +260,7 @@ struct Window : ZeroedMemoryAllocator {
 	const NWidgetCore *nested_focus; ///< Currently focused nested widget, or \c nullptr if no nested widget has focus.
 	std::map<int, QueryString*> querystrings; ///< QueryString associated to WWT_EDITBOX widgets.
 	NWidgetBase *nested_root;        ///< Root of the nested tree.
-	NWidgetBase **nested_array;      ///< Array of pointers into the tree. Do not access directly, use #Window::GetWidget() instead.
-	uint nested_array_size;          ///< Size of the nested array.
+	WidgetLookup widget_lookup; ///< Indexed access to the nested widget tree. Do not access directly, use #Window::GetWidget() instead.
```

Comment:
Could this now also be private (or protected)?

Seems all access has wrappers around it.

Reply:
DecreaseWindowCounters() accesses it, and that's static, so not quite.